edx_when/api.py

```diff
@@ -327,21 +394,14 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
             existing_date.active = True
         except models.ContentDate.DoesNotExist as error:
             if user:
+                # A UserDate creation below requires an existing ContentDate.
                 raise MissingDateError(block_id) from error
             existing_date = models.ContentDate(course_id=course_id, location=block_id, field=field)
-
-            # We had race-conditions create multiple DatePolicies w/ the same values. Handle that case.
-            try:
-                existing_policies = list(models.DatePolicy.objects.filter(**date_kwargs).order_by('id'))
-            except models.DatePolicy.DoesNotExist:
-                existing_policies = []
-            if existing_policies:
-                existing_date.policy = existing_policies[0]
-            else:
-                existing_date.policy = models.DatePolicy.objects.create(**date_kwargs)
+            _set_content_date_policy(date_kwargs, existing_date)
             needs_save = True
 
         if user and not user.is_anonymous:
+            # Course block date is for a particular user.
```

Comment:
Super Nit: Put the comment here and on line 419 above the if/else lines (one line higher) rather than just below. I personally always interpret python comments on what is directly beneath so seeing this comment and then the UserDate below, I was like yeah, cool, checks out. But then on line 419, the if statement didn't click with the comment and I realized it was applying to the whole else block. It's a small thing, feel free to ignore

Reply:
Will do.